02_bubble/davidbroughsmyth/main.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(s []int) []int {
+	cpy := make([]int, len(s))
+	copy(cpy, s)
+	for n := len(cpy); n > 1; n-- {
+		for i := 1; i < n; i++ {
```

Comment:
I'd suggest a better name for the loop indicies would be `i` and `j` as you have done in the insertion sort. `n` is often used as a count of some sort and is fixed for the duration of the loop. I was initially confused by the inner loop always counting up to `n` as I would typically expect something called `n` to be the length of the slice your iterating over. My confusion subsided rather quickly once I inspected the code more closely, but that would have been avoided by using `i` and `j`.